### PR TITLE
Fix failure when reading hidden columns in Hudi

### LINKED ISSUE #14341: Selecting hidden columns fails to open split in Hudi
We should throw column not found exception if we don't support them. 

```
SELECT "$path" FROM hudi.tests.hudi_cow_pt_tbl;

Query 20220928_074646_00017_n54ug failed: Error opening Hudi split file:/var/folders/8s/dkvf18z55lj_9yxhy1n54sph0000gn/T/TrinoTest15358788863231304430/data/hudi_cow_pt_tbl/dt=2021-12-09/hh=10/719c3273-2805-4124-b1ac-e980dada85bf-0_0-27-1215_20220906063435640.parquet (offset=0, length=435204): column type must be REGULAR: $path:string:SYNTHESIZED
io.trino.spi.TrinoException: Error opening Hudi split file:/var/folders/8s/dkvf18z55lj_9yxhy1n54sph0000gn/T/TrinoTest15358788863231304430/data/hudi_cow_pt_tbl/dt=2021-12-09/hh=10/719c3273-2805-4124-b1ac-e980dada85bf-0_0-27-1215_20220906063435640.parquet (offset=0, length=435204): column type must be REGULAR: $path:string:SYNTHESIZED
	at io.trino.plugin.hudi.HudiPageSourceProvider.createPageSource(HudiPageSourceProvider.java:268)
	at io.trino.plugin.hudi.HudiPageSourceProvider.createPageSource(HudiPageSourceProvider.java:165)
	at io.trino.plugin.base.classloader.ClassLoaderSafeConnectorPageSourceProvider.createPageSource(ClassLoaderSafeConnectorPageSourceProvider.java:49)
	at io.trino.split.PageSourceManager.createPageSource(PageSourceManager.java:62)
	at io.trino.operator.TableScanOperator.getOutput(TableScanOperator.java:308)
	at io.trino.operator.Driver.processInternal(Driver.java:411)
	at io.trino.operator.Driver.lambda$process$10(Driver.java:314)
	at io.trino.operator.Driver.tryWithLock(Driver.java:706)
	at io.trino.operator.Driver.process(Driver.java:306)
	at io.trino.operator.Driver.processForDuration(Driver.java:277)
	at io.trino.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:736)
	at io.trino.execution.executor.PrioritizedSplitRunner.process(PrioritizedSplitRunner.java:164)
	at io.trino.execution.executor.TaskExecutor$TaskRunner.run(TaskExecutor.java:515)
	at io.trino.$gen.Trino_testversion____20220928_074154_36.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: column type must be REGULAR: $path:string:SYNTHESIZED
	at com.google.common.base.Preconditions.checkArgument(Preconditions.java:220)
	at io.trino.plugin.hive.parquet.ParquetPageSourceFactory.createParquetReaderColumns(ParquetPageSourceFactory.java:440)
	at io.trino.plugin.hudi.HudiPageSourceProvider.createPageSource(HudiPageSourceProvider.java:227)
	... 16 more
```

Below columns cause this issue
* `$path`
* `$file_size`
* `$file_modified_time`
* `$partition`

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Fixes #14341

## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# Hudi
* Fix failure when reading hidden columns. ({issue}`14341`)
```

### CODE REVIEW

File: plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSource.java [hunk 1]
```diff
@@ -13,36 +13,87 @@
...
-    private final Map<String, Block> partitionBlocks;
 
     public HudiPageSource(
+            String partitionName,
             List<HiveColumnHandle> columnHandles,
             Map<String, Block> partitionBlocks,
-            ConnectorPageSource dataPageSource)
+            ConnectorPageSource dataPageSource,
+            Path path,
+            long fileSize,
```
Comment (codope) [inline plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSource.java:55]: Just for my understanding, how does file size within a page help? Is it used in table statistics as well?

File: plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSource.java [hunk 2]
```diff
@@ -13,36 +13,87 @@
...
+        int outputIndex = 0;
+        int delegateIndex = 0;
+
+        for (HiveColumnHandle column : columnHandles) {
+            if (partitionBlocks.containsKey(column.getName())) {
+                Block partitionValue = partitionBlocks.get(column.getName());
+                prefilledBlocks[outputIndex] = partitionValue;
+                delegateIndexes[outputIndex] = -1;
+            }
+            else if (column.getName().equals(PARTITION_COLUMN_NAME)) {
```
Comment (codope) [inline plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSource.java:74]: I also notice that there are few other internal columns in `HiveColumnHandle` such as `$bucket` and `$row_id`. When do we need to those columns?

**Review by findepi — APPROVED** [review]

File: plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSource.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSource.java:55]: The size exists for just providing `$file_size`. It unrelates to table statistics.

File: plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSource.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiPageSource.java:74]: The `$bucket` simply provides the bucket number of the current row. `$row_id` is an internal column for `UPDATE` and `MERGE` statements. The former is for users, latter is for internal usage.

**Pull request merged**